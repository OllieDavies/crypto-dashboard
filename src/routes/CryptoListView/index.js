import React, { useState } from "react";
import CryptoList from "../../components/CryptoList";
import OrderDropdown from "../../components/OrderDropdown";

import styles from "./styles.module.css";

const CryptoListView = () => {
  const [currentOrder, setOrder] = useState("desc");

  return (
    <div>
      <div className={styles.headingContainer}>
        <h1 className={styles.heading}>All Coins</h1>
        <OrderDropdown
          containerClassName={styles.dropdown}
          onChange={setOrder}
          options={{
            desc: "Price: High To Low",
            asc: "Price: Low To High",
          }}
        />
      </div>
      <CryptoList order={currentOrder} />
    </div>
  );
};

export default CryptoListView;
