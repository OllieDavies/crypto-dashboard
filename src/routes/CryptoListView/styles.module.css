.headingContainer {
  display: flex;
  align-items: center;
}

.heading {
  flex: 1;
}

.dropdown {
  text-align: right;
}