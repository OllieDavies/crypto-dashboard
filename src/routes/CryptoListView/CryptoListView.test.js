import React from "react";
import { render } from "@testing-library/react";

import CryptoListView from ".";

describe("<CryptoListView />", () => {
  it("matches snapshot", () => {
    const { container } = render(<CryptoListView />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
