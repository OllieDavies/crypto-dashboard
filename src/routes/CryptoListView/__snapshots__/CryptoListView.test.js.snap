// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoListView /> matches snapshot 1`] = `
<div>
  <div
    class="headingContainer"
  >
    <h1
      class="heading"
    >
      All Coins
    </h1>
    <select
      class="dropdown select"
      data-testid="orderdropdownSelect"
    >
      <option
        value="desc"
      >
        Price: High To Low
      </option>
      <option
        value="asc"
      >
        Price: Low To High
      </option>
    </select>
  </div>
  <h1>
    Loading...
  </h1>
</div>
`;
